crates/dc_bundle/src/definition/element.rs

```diff
@@ -366,3 +367,46 @@ impl Path {
         self
     }
 }
+
+// Implement the Eq and Hash traits so that ImageKey can be used as a hash table key
+impl Eq for ImageKey {}
```

Comment:
I am curious how this empty impl for the Eq work... Is it because it has a string field which implements Eq so we can just do this?

Reply:
For a Rust struct to implement `Eq`, it has to implement `PartialEq` also. `PartialEq` means you can compare them, like `a == b`, but it also means it may not be reflexive, like with floating point numbers `NaN != NaN`. So this implementation of `Eq` just relies on the `PartialEq` implementation, but also says that for any `ImageKey`, `a == a`. https://users.rust-lang.org/t/what-is-the-difference-between-eq-and-partialeq/15751